src/components/templates/Camp/components/RoomModal.tsx

```diff
@@ -46,14 +46,16 @@ export const RoomModal: React.FC<RoomModalProps> = ({
   const venue = useSelector(currentVenueSelector);
   const venues = useSelector(orderedVenuesSelector);
   const venueEvents = useSelector(venueEventsSelector) ?? [];
-  const users = usePartygoers();
+  const { recentWorldUsers } = useRecentWorldUsers();
   const venueName = venue?.name;
   const roomTitle = room?.title;
 
   const usersToDisplay = useMemo(
     () =>
-      users.filter((user) => user.lastSeenIn?.[`${venueName}/${roomTitle}`]),
-    [users, venueName, roomTitle]
+      recentWorldUsers.filter(
+        (user) => user.lastSeenIn?.[`${venueName}/${roomTitle}`]
+      ),
+    [recentWorldUsers, venueName, roomTitle]
   );
```

Comment:
This would be resolved by a small hook such as `recentUsersInRoom(venueId, roomName)`, which would probably consume `recentUsersInVenue(venueId)`